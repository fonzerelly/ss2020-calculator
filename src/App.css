.app {
  height: 100vh;
  width: 100vw;
  align-items:center;
  display: flex;
  justify-content: center;
}

.calculator {
  background-color: var(--calculator-background-color);
  height: var(--calculaotr-height);
  width: var(--calculator-width)
}

.display {
  align-items: center;
  background: var(--display-background-color);
  display: flex;
  height: var(--display-height);
  padding: 0 4%;
  width: var(--display-width);
}

.display__value {
  color: var(--display-text-color);
  font-size: var(--display-text-size);
  font-family: var(--main-font);
  font-weight: var(--display-text-weight);
  margin-left: auto;
  overflow: hidden;
}

.keypad {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  height: var(--keypad-height);
  padding: 2%;
  width: var(--keypad-width);
}

.keypad__number {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  height: 80%;
  width: 75%;
}

.keypad__operator {
  display: flex;
  flex-direction: column;
  height: 80%;
  width: 25%;
}

.keypad__submit {
  height: 20%;
  width: 100%;
}

.keypad__key {
  align-items: center;
  display: flex;
  height: 25%;
  justify-content: center;
  transition: background-color 0.3s linear;
}

.keypad__key:hover {
  cursor: pointer;
}

.keypad__key--operator {
  background-color: var(--action-key-color);
  width: 100%;
}

.keypad__key--operator:hover {
  background-color: var(--action-key-color-hover);
}

.keypad__key--number {
  background-color: var(--number-key-color);
  width: calc(100%/3);
}

.keypad__key--number:hover {
  background-color: var(--number-key-color-hover);
}

.keypad__key--submit {
  background-color: var(--submit-key-color);
  height: 100%;
  width: 100%;
}

.keypad__key--submit:hover {
  background-color: var(--submit-key-color-hover);
}

.keypad__key__value {
  color: var(--key-text-color);
  font-family: var(--main-font);
  font-size: var(--key-text-size);
  font-weight: var(--key-text-weight);
  text-align: center;
}