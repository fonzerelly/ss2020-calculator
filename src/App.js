import React from 'react';
import './App.css';

import {CalculatorProvider} from './calculator-context'
import Display from './display'
import Numpad from './numpad'
import Opkey from './operatorkey'
import Evalkey from './evaluatekey'

function App() {
  return (
    <CalculatorProvider>
    <div className="app">
      <div className="calculator">
        <Display/>
        <div className="keypad">
          <Numpad/>
          <div className="keypad__operator">
            <Opkey label="/"/>
            <Opkey label="*"/>
            <Opkey label="+"/>
            <Opkey label="-"/>
          </div>
          <Evalkey/>
        </div>
      </div>
    </div>
    </CalculatorProvider>
  );
}

export default App;
