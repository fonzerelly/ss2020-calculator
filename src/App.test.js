import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import renderer from 'react-test-renderer'
describe('App', () => {
  it('render as always', () => {
    const tree = renderer.create(<App/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
  })
})