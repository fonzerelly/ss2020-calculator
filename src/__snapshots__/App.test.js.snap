// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render as always 1`] = `
<div
  className="app"
>
  <div
    className="calculator"
  >
    <div
      className="display"
    >
      <p
        className="display__value"
      >
        0
      </p>
    </div>
    <div
      className="keypad"
    >
      <div
        className="keypad__number"
      >
        <div
          className="keypad__key--number"
        >
          <p
            className="keypad__key__value"
          >
            9
          </p>
        </div>
        <div
          className="keypad__key--number"
        >
          <p
            className="keypad__key__value"
          >
            8
          </p>
        </div>
        <div
          className="keypad__key--number"
        >
          <p
            className="keypad__key__value"
          >
            7
          </p>
        </div>
        <div
          className="keypad__key--number"
        >
          <p
            className="keypad__key__value"
          >
            6
          </p>
        </div>
        <div
          className="keypad__key--number"
        >
          <p
            className="keypad__key__value"
          >
            5
          </p>
        </div>
        <div
          className="keypad__key--number"
        >
          <p
            className="keypad__key__value"
          >
            4
          </p>
        </div>
        <div
          className="keypad__key--number"
        >
          <p
            className="keypad__key__value"
          >
            3
          </p>
        </div>
        <div
          className="keypad__key--number"
        >
          <p
            className="keypad__key__value"
          >
            2
          </p>
        </div>
        <div
          className="keypad__key--number"
        >
          <p
            className="keypad__key__value"
          >
            1
          </p>
        </div>
        <div
          className="keypad__key--number"
        >
          <p
            className="keypad__key__value"
          >
            .
          </p>
        </div>
        <div
          className="keypad__key--number"
        >
          <p
            className="keypad__key__value"
          >
            0
          </p>
        </div>
        <div
          className="keypad__key--number"
        >
          <p
            className="keypad__key__value"
          >
            ce
          </p>
        </div>
      </div>
      <div
        className="keypad__operator"
      >
        <div
          className="keypad__key--operator"
        >
          <p
            className="keypad__key__value"
          >
            /
          </p>
        </div>
        <div
          className="keypad__key--operator"
        >
          <p
            className="keypad__key__value"
          >
            *
          </p>
        </div>
        <div
          className="keypad__key--operator"
        >
          <p
            className="keypad__key__value"
          >
            +
          </p>
        </div>
        <div
          className="keypad__key--operator"
        >
          <p
            className="keypad__key__value"
          >
            -
          </p>
        </div>
      </div>
      <div
        className="keypad__submit"
      >
        <div
          className="keypad__key--submit"
        >
          <p
            className="keypad__key__value"
          >
            =
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
